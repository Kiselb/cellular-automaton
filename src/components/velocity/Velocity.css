.velocity {
    width: 100%;
    box-sizing: border-box;
    font-size: 1em;
    border-top-style: hidden;
    border-right-style: hidden;
    border-left-style: hidden;
    border-bottom: 1px solid lightgray;
    outline: none;
}
