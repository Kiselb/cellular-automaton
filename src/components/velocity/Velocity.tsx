import React from "react";

import "./Velocity.css";

type VelocityItem = {
  delay: number;
  name: string;
};
export type VelocityHandler = (delay: number) => void;
export type VelocityProps = {
  onVelocityChange: VelocityHandler;
  defVelocity: number;
};

const VelocitySet: VelocityItem[] = [
  { delay: 1000, name: "Медленно" },
  { delay: 2000, name: "Очень медленно" },
  { delay: 250, name: "Быстро" },
  { delay: 100, name: "Очень быстро" },
  { delay: 500, name: "Нормально" },
];

export const Velocity = ({
  onVelocityChange,
  defVelocity,
  ...props
}: VelocityProps) => {
  const onChange = (event: React.FormEvent<HTMLSelectElement>) => {
    event.preventDefault();
    onVelocityChange(parseInt(event.currentTarget.value));
  };
  return (
    <select
      className="velocity"
      data-testid="velocity"
      onChange={onChange}
      defaultValue={defVelocity}
    >
      {VelocitySet.sort((a, b) => {
        if (a.delay < b.delay) return -1;
        if (a.delay > b.delay) return 1;
        return 0;
      }).map((item, index) => {
        return (
          <option key={index} value={item.delay}>
            {item.name}
          </option>
        );
      })}
    </select>
  );
};
