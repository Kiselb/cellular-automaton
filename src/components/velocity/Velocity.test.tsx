import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { Velocity } from "./Velocity";

it("Control Velocity renders correctly", () => {
  const { asFragment } = render(
    <Velocity onVelocityChange={() => alert("OK")} defVelocity={100} />
  );
  screen.debug();
  expect(asFragment()).toMatchSnapshot();
});
it("Control Velocity has style", () => {
  render(<Velocity onVelocityChange={() => alert("OK")} defVelocity={100} />);
  const control = screen.getByTestId("velocity");
  expect(control).toHaveClass("velocity");
});
it("Control Velocity fire event", () => {
  const probe = jest.fn();
  render(<Velocity onVelocityChange={probe} defVelocity={100} />);
  const control = screen.getByTestId("velocity");
  fireEvent.change(control);
  expect(probe).toBeCalled();
});
