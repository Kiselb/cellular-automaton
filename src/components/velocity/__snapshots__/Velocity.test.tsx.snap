// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Velocity renders correctly 1`] = `
<DocumentFragment>
  <select
    class="velocity"
    data-testid="velocity"
  >
    <option
      selected=""
      value="100"
    >
      Очень быстро
    </option>
    <option
      value="250"
    >
      Быстро
    </option>
    <option
      value="500"
    >
      Нормально
    </option>
    <option
      value="1000"
    >
      Медленно
    </option>
    <option
      value="2000"
    >
      Очень медленно
    </option>
  </select>
</DocumentFragment>
`;
